scmrepo/git/backend/pygit2.py

```diff
@@ -317,13 +317,16 @@ def set_ref(
             )
 
     def get_ref(self, name, follow: bool = True) -> Optional[str]:
-        from pygit2 import GIT_REF_SYMBOLIC
+        from pygit2 import GIT_OBJ_COMMIT, GIT_REF_SYMBOLIC, Tag
 
         ref = self.repo.references.get(name)
         if not ref:
             return None
         if follow and ref.type == GIT_REF_SYMBOLIC:
             ref = ref.resolve()
+        commit, _ = self.repo.resolve_refish(str(ref.target))
+        if isinstance(commit, Tag):
+            return str(commit.peel(GIT_OBJ_COMMIT).id)
```

Comment:
I added annotated tag to `test_get_ref` and `pygit2` was failing. 
Not sure if this fix makes sene

Reply:
We shouldn't be using `resolve_refish` in `get_ref`, we've already resolved the ref and have the resolved ID for the object it points to (`ref.target`), so we can get the object directly from `self.repo`. Also, we already have a wrapper for `resolve_refish` that peels tags: https://github.com/iterative/scmrepo/blob/4253fe234d1a26a821356d4121accac11f24a86c/scmrepo/git/backend/pygit2.py#L110

(But we don't want to use `_resolve_refish` as a replacement for `get_ref`, since it will still match when `name` is a branch/tag shortname, or something that looks like a git SHA)

This fix should probably be something like

```suggestion
        obj = self.repo[ref.target]
        if isinstance(obj, Tag):
            return str(obj.peel(GIT_OBJ_COMMIT).id)
```

(this is untested, see the pygit docs for `references` and `objects` https://www.pygit2.org/references.html if it doesn't work)